wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -1622,14 +1627,21 @@ protected JSch getJSch(final OpenSshConfig.Host hc, FS fs) throws JSchException
                     String password = remoteUrl.substring(remoteUrl.lastIndexOf(":") + 1, remoteUrl.indexOf("@"));
                     command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, password));
                 } else {
-                    credentials = credentialsLoader.getUserCredential(remoteUrl);
-                    if (credentials != null) {
-                        command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(credentials.getUserName(),
-                                                                                               credentials.getPassword()));
+                    String gitUser = request.getAttributes().get("username");
+                    String gitPassword = request.getAttributes().get("password");
+                    if (gitUser != null && gitPassword != null) {
+                        setCurrentCredentials(gitUser, gitPassword);
```

Comment:
You should remove it because you already set credentials for command. So GitBasicAuthenticationCredentialsProvider usage is useless here